monitoring/mongodb/dashboard.py

```diff
@@ -0,0 +1,1397 @@
+    "mongos": "zenko/data-db-mongodb-sharded-mongos",
+    "shards": "zenko/data-db-mongodb-sharded-shards-data",
+    "shard_pattern": "zenko/data-db-mongodb-sharded-shard.*-data",
+    "all_shards": ".*shards.*",
+}
+
+# Common thresholds
+THRESHOLDS = {
+    "green_red": [
+        Threshold("green", 0, 0.0),
+        Threshold("red", 1, 80.0),
+    ],
+    "oplog_window": [
+        Threshold("red", 0, 0.0),
+        Threshold("orange", 1, 600.0),
+        Threshold("green", 2, 900.0),
+    ],
+    "disk_usage": [
+        Threshold("green", 0, 0.0),
+        Threshold("orange", 1, 70.0),
+        Threshold("red", 2, 90.0),
+    ],
+    "connections": [
+        Threshold("green", 0, 0.0),
+        Threshold("yellow", 1, 300.0),
+        Threshold("orange", 2, 400.0),
+        Threshold("red", 3, 500.0),
+    ],
+    "latency": [
+        Threshold("green", 0, 0.0),
+        Threshold("yellow", 1, 1000.0),
+        Threshold("orange", 2, 5000.0),
+        Threshold("red", 3, 100000.0),
+    ],
+    "queue_ops": [
+        Threshold("green", 0, 0.0),
+        Threshold("orange", 1, 1.0),
+        Threshold("red", 2, 2.0),
+    ],
+    "query_exec": [
+        Threshold("green", 0, 0.0),
+        Threshold("green", 1, 1000.0),
+        Threshold("yellow", 2, 10000.0),
+        Threshold("orange", 3, 100000.0),
+        Threshold("red", 4, 1000000.0),
+    ],
+    "write_conflicts": [
+        Threshold("green", 0, 0.0),
+        Threshold("orange", 1, 5.0),
+        Threshold("red", 2, 10.0),
+    ],
+    "replication_lag": [
+        Threshold("transparent", 0, 0.0),
+        Threshold("red", 1, 10.0),
+    ],
+}
+
+# MongoDB State Mappings for StateTimeline - create a single StatValueMappings with all items
+MONGODB_STATES = [
+    core.StatValueMappings(
+        core.StatValueMappingItem(text="STARTUP", mapValue="0", color="", index=11),
+        core.StatValueMappingItem(text="PRIMARY", mapValue="1", color="blue", index=4),
+        core.StatValueMappingItem(text="SECONDARY", mapValue="2", color="purple", index=3),
+        core.StatValueMappingItem(text="RECOVERING", mapValue="3", color="", index=10),
+        core.StatValueMappingItem(text="STARTUP2", mapValue="4", color="", index=12),
+        core.StatValueMappingItem(text="STARTUP2", mapValue="5", color="", index=9),
+        core.StatValueMappingItem(text="UNKNOWN", mapValue="6", color="", index=8),
+        core.StatValueMappingItem(text="ARBITER", mapValue="7", color="#8B8000", index=1),
+        core.StatValueMappingItem(text="DOWN", mapValue="8", color="", index=7),
+        core.StatValueMappingItem(text="ROLLBACK", mapValue="9", color="", index=6),
+        core.StatValueMappingItem(text="REMOVED", mapValue="10", color="", index=5),
+        core.StatValueMappingItem(text="Exporter is not connected", mapValue="null", color="#FF7383", index=2),
+    )
+]
+
+
+class MongoDBMetrics:
+    """MongoDB metrics expressions"""
```

Comment:
adding this here, as it applies to many metrics : we should properly filter the "cluster" of mongo, as there may be multiple in the cluster → should be done based on the params passed by Zenko (and possibly adding template params, typically in case a single Zenko instance deploys multiple MongoDB clusters - but we are not there yet AFAIK)

this can be done easily using with the Metrics classes, as this can be set as default filters in the definition

Reply:
you're absolutely right, we do not need to support multiple clusters yet.

my point was just that we should start (today) with a simple filtering (supporting the single cluster we have now):
```
   VERSION_INFO=Metrics('mongodb_version_info', 'job=~"{Jobs.shards}|{Jobs.configsvr}|{Jobs.mongos}', namespace=namspace)
```

and this can **later** be extended to support.